flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/utils/ContinuousUnBoundingSettings.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.jdbc.utils;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.io.Serializable;
+import java.time.Duration;
+import java.util.Objects;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Settings describing how to do continuous file discovery and enumeration for the file source's
+ * continuous discovery and streaming mode.
+ */
+@PublicEvolving
```

Comment:
```suggestion
@Experimental
```

We should use `Experimental` for the first version, and update it to `PublicEvolving` after it's stable.

Reply:
Thanks @1996fanrui for the review.

<img width="1419" alt="image" src="https://github.com/apache/flink-connector-jdbc/assets/64569824/2ad4d5a4-8bf7-4987-8e18-ca079c7a59c1">

I sorted out the rules based on the archunit of flink repo.

- For config options:
`Experimental` annatation is allowed to use.


- For table api:
`Experimental` annatation is allowed to use.

- For ApiAnnotation:
`Experimental` annotation is allowed to use for classes/API that is located in `org.apache.flink..api..` package and  not located in `..internal.. ` pacakge 

I made two draft pr https://github.com/apache/flink-connector-jdbc/pull/125  https://github.com/apache/flink-connector-jdbc/pull/124  in using `Experimental` for checking rule links.  It was failed.

So, It seems that the `Experimental` annotation is not suitable here, just `PublicEvolving`
please let me know what's your opinion.

Thanks a lot.